django/db/models/base.py

```diff
@@ -481,7 +481,7 @@ def __init__(self, *args, **kwargs):
                 except (AttributeError, FieldDoesNotExist):
                     pass
             for kwarg in kwargs:
-                raise TypeError("'%s' is an invalid keyword argument for this function" % kwarg)
+                raise TypeError("%s() got an unexpected keyword argument '%s'" % (cls.__name__, list(kwargs)[0]))
```

Comment:
Did you intend to change `kwarg` to `list(kwargs)[0]` and keep the loop?

Reply:
Sorry, no, I made the patch over an earlier codebase and didn’t notice the change. Should be kwarg.